### PR TITLE
feat(check_dep_sync): compare version ranges between pixi.toml and pyproject.toml

### LINKED ISSUE #497: Sync check should compare pixi.toml ranges, not just assert bounds exist
scripts/check_dep_sync.py currently only checks that each [project.dependencies] entry has a < upper bound. It does not verify that the version ranges match pixi.toml [pypi-dependencies]. A maintainer could update pixi.toml without updating pyproject.toml and the script would not catch the drift. Enhance the script to parse both files and assert that each package's specifier in pyproject.toml matches the one in pixi.toml.

_Follow-up from #338_

### PR DESCRIPTION
**Pull request by mvillmow**
## Summary
Enhanced check_dep_sync.py to verify that version ranges in pyproject.toml match those in pixi.toml, preventing drift when either file is updated without synchronizing the other.

## Changes
- Parse both pixi.toml and pyproject.toml to extract version specifiers
- Compare version ranges for each dependency between the two files
- Fail if version ranges don't match, catching unsynchronized updates

## Testing
- Added comprehensive test suite in tests/test_check_dep_sync.py covering version range matching, mismatches, and edge cases

## Closes
Closes #497

Generated by Claude Code via ProjectHephaestus automation.

### CODE REVIEW

File: coverage.xml [hunk 1]
```diff
@@ -0,0 +1,855 @@
+<?xml version="1.0" ?>
```
Comment (mvillmow) [inline coverage.xml]: This 855-line `coverage.xml` is a generated test-coverage artifact that should not be committed. It is unrelated to issue #497, and it embeds an absolute machine-specific path (`<source>/home/mvillmow/Projects/ProjectHermes/build/.worktrees/issue-497</source>`) that will be wrong for every other checkout. `.gitignore` already ignores `.coverage` but not `coverage.xml`. Fix: `git rm --cached coverage.xml` and add `coverage.xml` (and `htmlcov/` if not already covered) to `.gitignore`.

**Review by mvillmow — COMMENTED** [review]
Feature code (bidirectional check_parity + tests) is correct and complete for #497, but PR commits a generated 855-line coverage.xml with an embedded local path — remove and gitignore before merge.
